tests/resources/test_item.py

```diff
@@ -1653,3 +1653,67 @@ async def test_get_filter_extension(app_client, load_test_data, load_test_collec
     fc = resp.json()
     assert len(fc["features"]) == 1
     assert fc["features"][0]["id"] == search_id
+
+
+async def test_post_filter_extension(app_client, load_test_data, load_test_collection):
+    """Test POST with Filter extension"""
+    test_item = load_test_data("test_item.json")
+    collection_id = test_item["collection"]
+    ids = []
+
+    # Ingest 5 items
+    for _ in range(5):
+        uid = str(uuid.uuid4())
+        test_item["id"] = uid
+        resp = await app_client.post(
+            f"/collections/{collection_id}/items", json=test_item
+        )
+        assert resp.status_code == 201
+        ids.append(uid)
+
+    search_id = ids[2]
+
+    # SEARCH
+    # CQL2-JSON
+    resp = await app_client.post(
+        "/search",
+        json={
+            "filter-lang": "cql2-json",
+            "filter": {"op": "in", "args": [{"property": "id"}, [search_id]]},
+        },
+    )
+    assert resp.status_code == 200
+    fc = resp.json()
+    assert len(fc["features"]) == 1
+    assert fc["features"][0]["id"] == search_id
+
+    # CQL-JSON
+    resp = await app_client.post(
+        "/search",
+        json={
+            "filter-lang": "cql-json",
+            "filter": {
+                "eq": [
+                    {"property": "id"},
+                    search_id,
+                ],
+            },
+        },
+    )
+    assert resp.status_code == 200
+    fc = resp.json()
+    assert len(fc["features"]) == 1
+    assert fc["features"][0]["id"] == search_id
+
+    # CQL2-TEXT
+    resp = await app_client.post(
+        "/search",
+        json={
+            "filter-lang": "cql2-text",
+            "filter": f"id='{search_id}'",
+        },
+    )
+    assert resp.status_code == 200
+    fc = resp.json()
+    assert len(fc["features"]) == 1
+    assert fc["features"][0]["id"] == search_id
```

Comment:
This is failing right now because pydantic expect a dictionary not a string

Reply:
Oo 

https://github.com/stac-api-extensions/filter?tab=readme-ov-file#conformance-classes

> If both are advertised as being supported, it is only required that both be supported for GET query parameters, and that only that CQL2 JSON be supported for POST JSON requests. It is recommended that clients use CQL2 Text in GET requests and CQL2 JSON in POST requests.

@hrodmn I don't thin the POST request needs to support cql2-text so in fact it's a simpler fix we just need to update https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/request.py#L73

Reply:
So we just need to make sure `filter-lang="cql2-text"` is not allowed? I realize that `cql2-text` in POST requests is not required by the spec so it's definitely not necessary but maybe convenient in some cases.

Reply:
yes

Reply:
FYI: I reverted this, and removed the `cql2-text` tests for POST endpoint because it's not needed